peps/pep-0750.rst

```diff
@@ -414,8 +414,46 @@ except that all expressions are always implicitly wrapped with a ``lambda``:
     lexical scope, including local and global variables. Any valid Python expression
     can be used, including function and method calls.
 
-This means that the lambda wrapping here uses the usual lexical scoping. As with
-f-strings, there's no need to use ``locals()``, ``globals()``, or frame
+However, this PEP further proposes to extend the usual lexical scoping of a lambda-wrapped
+expression to that supported by
+`annotation scopes <https://docs.python.org/3/reference/executionmodel.html#annotation-scopes>`_,
+so as to minimize developer surprise. For the remainder of the section, *class scope* will be
+used to refer to annotation scope as it's used by tag strings; this naming will be discussed
+momentarily.
+
+Let's look at why this is important with a somewhat contrived example where a developer sets up
+how figures are rendered in a caption, using a ``html`` tag:
+
+.. code-block:: python
+
+    class SomeConfig:
+        tag = 'b'
+        figure = html'<{tag}>Figure</{tag}>'
+
+If the scope is only the usual lexical scope for lambdas, when ``figure`` is finally evaluated,
+the name ``tag`` is not available. Contrast that scoping with what is seen in using a f-string:
+
+.. code-block:: python
+
+    class SomeConfig:
+        tag = 'b'
+        figure = f'<{tag}>Figure</{tag}>'
+
+The class variable ``figure`` here does evaluate correctly, if at the risk of an HTML injection
+attack. Because tag strings are supposed to work like f-strings, but with more capabilities,
+it's necessary to support class scope.
+
+Why is it called class scope, instead of annotation scope?
```

Comment:
I see the argument for not using "annotation scope", though I'd note that annotation scopes are already used for non-annotation use cases (e.g., type alias statements), but surely "class scope" is an even more confusing term: you'd think it would refer to the scope of a class, which does *not* follow the same rules as an annotation scope. The documentation for annotation scopes (https://docs.python.org/3.12/reference/executionmodel.html#annotation-scopes) explicitly talks about class scopes as a different concept.

Reply:
Naming things... let's see what we can do here.

Reply:
@JelleZijlstra perhaps *class definition* scope would be better? Again, this should be considered to be a synonym for annotation scope, but without having to explain that annotation scope is more than just for annotations, a possible point of confusion (as it was for me).

Or we could just explain that annotation scope does what we need here, and move on.

Reply:
Personally I'd prefer to stick with "annotation scope" here, since that term is already in the documentation. We can separately discuss renaming the concept, but that could be an independent change from this PEP.

Reply:
Changed to "annotation scope"